suite-native/firmware/src/screens/FirmwareUpdateInProgressScreen.tsx

```diff
@@ -106,12 +120,17 @@ export const FirmwareUpdateInProgressScreen = () => {
                 // Action cancelled on device
                 result.payload?.code === 'Failure_ActionCancelled'
             ) {
+                handleAnalyticsReportCancelled();
                 navigation.navigate(DeviceStackRoutes.FirmwareUpdate);
             }
 
+            handleAnalyticsReportFinished({ error: result.payload?.error });
```

Comment:
What about adding here some default error message so it doesn't look like a success in analytics if result.payload?.error is empty?

Reply:
This is success in case it's empty

Reply:
Good idea